src/jdk.jpackage/macosx/classes/jdk/jpackage/internal/MacApplicationBuilder.java

```diff
@@ -123,6 +125,17 @@ private static void validateAppVersion(Application app) throws ConfigException {
         }
     }
 
+    private static void validateAppContentDirs(Application app) {
+        for (var contentDir : app.contentDirs()) {
+            if (!CONTENTS_SUB_DIRS.stream()
```

Comment:
You could replace `CONTENTS_SUB_DIRS` with a `Set<String>`, and the inspection with `CONTENTS_SUB_DIRS.contains(contentDir.getFileName().toString())`.

Reply:
Updated as suggested.